tilavarauspalvelu/management/commands/data_creation/create_reservation_units.py

```diff
@@ -1,448 +1,1632 @@
+        ReservableWindowInfo(name="later", minimum=3, maximum=30),
+    ]
+    start_interval_choices: list[StartIntervalInfo] = [
+        StartIntervalInfo(name="15", value=ReservationStartInterval.INTERVAL_15_MINUTES),
+        StartIntervalInfo(name="30", value=ReservationStartInterval.INTERVAL_30_MINUTES),
+        StartIntervalInfo(name="90", value=ReservationStartInterval.INTERVAL_90_MINUTES),
+    ]
+    cancellation_rule_choices: list[CancelInfo] = [
+        CancelInfo(name="cannot", value=[None]),
+        CancelInfo(name="allowed", value=cancellation_rules),
+    ]
+    handling_info_choices: list[HandlingInfo] = [
+        HandlingInfo(name="no", handling_required=False),
+        HandlingInfo(name="yes", handling_required=True),
+    ]
+
+    ReservationUnitSpacesThoughModel: type[models.Model] = ReservationUnit.spaces.through  # noqa: N806
+
+    spaces: list[Space] = []
+    reservation_units: list[ReservationUnit] = []
+    reservation_unit_spaces: list[models.Model] = []
+    pricings: list[ReservationUnitPricing] = []
+    images: list[ReservationUnitImage] = []
+
+    reservation_unit_type = ReservationUnitTypeFactory.create(
+        name="Ilmainen",
+        name_fi="Ilmainen",
+        name_en="Free",
+        name_sv="Gratis",
+    )
 
+    # One shared unit for all free reservation units
+    unit = UnitFactory.create(
+        name="Keskustoimisto",
+        name_fi="Keskustoimisto",
+        name_en="Central office",
+        name_sv="Centrala kontoret",
+        tprek_id=None,
+        tprek_department_id=None,
+    )
+
+    unit_counter = itertools.count(start=1)
+
+    for data in get_combinations(
+        iterables=[
+            buffer_time_choices,
+            reservation_time_choices,
+            min_max_days_before_choices,
+            start_interval_choices,
+            cancellation_rule_choices,
+            handling_info_choices,
+        ],
+        output_type=FreeReservationUnitData,
+    ):
+        set_name: SetName = random.choice(SetName.non_free_of_charge_applying())
+        reservation_kind = ReservationKind.DIRECT
+        authentication = AuthenticationType.WEAK
+        number = next(unit_counter)
+
+        space = SpaceFactory.build_for_bulk_create(
+            name=f"Toimistokoppi #{number}",
+            name_fi=f"Toimistokoppi #{number}",
+            name_en=f"Office cubicle #{number}",
+            name_sv=f"Kontorsbås #{number}",
+            unit=unit,
+        )
+        spaces.append(space)
+
+        reservation_unit = (
+            ReservationUnitBuilder()
+            .set(
+                name=f"Ankea toimistokoppi #{number}",
+                name_fi=f"Ankea toimistokoppi #{number}",
+                name_en=f"Drab office cubicle #{number}",
+                name_sv=f"Dyster kontorsbås #{number}",
+                unit=unit,
+                origin_hauki_resource=random.choice(hauki_resources),
+                allow_reservations_without_opening_hours=True,
```

Comment:
Is True for `allow_reservations_without_opening_hours` needed anymore if these have `origin_hauki_resource` assigned?

Reply:
Probably not